packages/transformer/src/IModelExporter.ts

```diff
@@ -580,7 +596,7 @@ export class IModelExporter {
   public async exportCodeSpecByName(codeSpecName: string): Promise<void> {
     const codeSpec: CodeSpec = this.sourceDb.codeSpecs.getByName(codeSpecName);
     let isUpdate: boolean | undefined;
-    if (undefined !== this._sourceDbChanges) {
+    if (this._sourceDbChanges !== undefined && !this._sourceDbChanges.isEmpty) {
```

Comment:
[Q] Can you explain more about this change: does this change means that all codespecs will be exported when sourceDbChanges are empty? Why we want that?
Is it possible to instead add needed code Specs to sourceDbChanges,  when custom changes are added? Same as e.g. parent models would be marked as updated when new model is inserted.

Reply:
Same question for other places like exportModelContainer()

Reply:
Originally I was doing this because I was always defining sourceDbChanges, but you're right that this is a logical error. I didnt understand why tests weren't failing ( because we have some that assert no changes) until I realized there are multiple places we can stop ourselves from exporting changes and I was getting saved by those future checks. Created work item to improve that test

Ive stopped using isEmpty except for in the transformer and sourceDbChanges can now be undefined if exporter is initialized with {}. 

```
 private async _tryInitChangesetData(args?: ProcessChangesOptions) {
    if (
      !args ||
      this.sourceDb.iTwinId === undefined ||
      this.sourceDb.changeset.index === undefined
    ) {
      this._sourceChangeDataState = "unconnected";
      return;
    }

    const noChanges =
      this.synchronizationVersion.index === this.sourceDb.changeset.index;
    if (noChanges) {
      this._sourceChangeDataState = "no-changes";
      this._csFileProps = [];
      return;
    }
```

I think probably in the case where no changesets and args are passed i.e. argsForProcessChanges, we will just set csFileProps to [] and sourceDbChanges will be defined, because we'll pass [] to exporter.initialize in transformer.